### ISSUE TITLE
How to use nodejs_image with fine grained deps

### ISSUE THREAD

**Issue by Toxicable** [bazelbuild/rules_docker#545]
Fine Grained Dependancies example here: https://github.com/alexeagle/angular-bazel-example/commit/6ab8b999e7dbb90621f0e4cd5037fa8612e0555a

Below is my rules setup for to test this.
As you can see the top layer (at 154.1MB) includes much more than jsut my src since that's only a few kb. I assume that it includes the NPM deps somehow.

Since I don't have to declare my `//libs/...` to `nodejs_image` I assume it collects up the deps from the `ts_library` rule which includes the NPM deps, I don't think is the desired behaviour here.

Ideas for how we're meant to be using fine grain deps?

```
NPM_DEPS = [
  "@npm//rxjs",
  "@npm//sequelize",
]

ts_library(
    name = "ts_server",
    deps = [ "//libs/common" ] + NPM_DEPS,
    srcs = glob(["*.ts", "src/**/*.ts"], exclude = ["src/**/*.spec.ts"]),
)
```
```
nodejs_image(
    name = "image.nodejs_image",
    entry_point = "__main__/apps/app.js",
    node_modules = "//:empty_node_modules",
    data = [":ts_server"],
    layers = NPM_DEPS,
)
```
`bazel run image.nodejs_image` has the output:
```
19c0ca213193: Loading layer [=>]    108MB/108MB
54096fada491: Loading layer [=>]  10.24kB/10.24kB
84ff92691f90: Loading layer [=>]  10.24kB/10.24kB
7c202ad0c165: Loading layer [=>]   9.39MB/9.39MB
3f15e2624e13: Loading layer [=>]  5.233MB/5.233MB
60023573fd80: Loading layer [=>]  154.1MB/154.1MB
````

where as this:
```
nodejs_image(
    name = "image.nodejs_image",
    entry_point = "__main__/apps/app.js",
    node_modules = "//:empty_node_modules",
    data = [":ts_server"] + NPM_DEPS,
)
```
`bazel run image.nodejs_image` has the output:
```
19c0ca213193: Loading layer =>]    108MB/108MB
54096fada491: Loading layer [=>]  10.24kB/10.24kB
84ff92691f90: Loading layer [=>]  10.24kB/10.24kB
60023573fd80: Loading layer [=>]  154.1MB/154.1MB
```

**Comment by xingao267** [comment]
@Toxicable, thanks for filing the issue. We recently have https://github.com/bazelbuild/rules_docker/pull/487 under review. Is that one helpful in your use case?

**Comment by Toxicable** [comment]
@xingao267 ah yes, that looks like it should probably solve the underlying issue here.
Ill leave this open till we can get that through.

**Comment by xingao267** [comment]
@Toxicable, sounds good, thanks!

**Comment by wtgtybhertgeghgtwtg** [comment]
Was this resolved by #1346?
